### PR TITLE
CI matrix change: Add OSX 12 arm

### PR DESCRIPTION
**Pull request by aik-jahoda**
Changes in CI matrix:

| OS | What to do | Notes
| - | - | -
| macOS 12.00 arm | Scout it (at least)  | 

part of #57947

### CODE REVIEW

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml [hunk 1]
```diff
@@ -107,7 +107,7 @@ jobs:
     # OSX arm64
     - ${{ if eq(parameters.platform, 'OSX_arm64') }}:
       - ${{ if and(eq(variables['System.TeamProject'], 'public'), in(parameters.jobParameters.helixQueueGroup, 'pr', 'ci', 'libraries')) }}:
-        - OSX.1100.ARM64.Open
+        - OSX.1500.ARM64.Open
```
Comment (danmoseley) [inline eng/pipelines/coreclr/templates/helix-queues-setup.yml:110]: What is 1500? I assumed that on Arm64, macOS followed the same version numbering -- ie macOS 11 or macOS12. Is that not right?

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 2]
```diff
@@ -90,6 +90,7 @@ jobs:
     # OSX arm64
     - ${{ if eq(parameters.platform, 'OSX_arm64') }}:
       - OSX.1100.ARM64.Open
+      - OSX.1200.ARM64.Open
```
Comment (danmoseley) [inline eng/pipelines/libraries/helix-queues-setup.yml:93]: Just curious, how can I tell this is libraries tests, as opposed eg to CLR tests ? It's not clear to me - I'm not familiar with how this works.

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml (hunk 1 above)
Comment (aik-jahoda) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:110]: Yeah, it is mistake, it should be 1200

**Comment by aik-jahoda** [comment]
@ViktorHofer, @danmoseley, all test passes, we can add this queue once we have more resources on Helix side.

File: eng/pipelines/libraries/helix-queues-setup.yml (hunk 2 above)
Comment (aik-jahoda) [reply eng/pipelines/libraries/helix-queues-setup.yml:93]: Ad far as I understand
CLR: eng/pipelines/coreclr/templates/helix-queues-setup.yml
Libs: eng/pipelines/libraries/helix-queues-setup.yml 

But it is still black box for me how it is orchestrated.

@ViktorHofer is there any docs available?

**Review by Anipik — APPROVED** [review]

File: eng/pipelines/libraries/helix-queues-setup.yml (hunk 2 above)
Comment (steveisok) [reply eng/pipelines/libraries/helix-queues-setup.yml:93]: This file associates the queue to platform. To see if it runs, you'll have to check the runtime ymls to see if the platform is defined.  A quick spot check looks like osx-arm64 is defined only for the runtime tests.

**Comment by steveisok** [comment]
> I didn't check the yml. Does this added queue only run as part of a rolling build? If so, can you please queue up a build with your changes to make sure that the changes work as expected?

Looks to only run on the rolling build

https://github.com/dotnet/runtime/blob/8bba6fb99e9579e8eb625efce1e1a7c74ac66ab7/eng/pipelines/runtime-staging.yml#L497

**Comment by aik-jahoda** [comment]
We are waiting for available HW in the CI queue.

**Comment by karelz** [comment]
@ericstj whom should we reassign it to, to track the new HW availability and then merge?

**Comment by ericstj** [comment]
Looks like https://github.com/dotnet/core-eng/issues/13921 tracks availability.  @aik-jahoda, do we have a recent update?  Looks like the last info was 1.5 months ago, it's possible things have changed.  Perhaps @MattGal or @ilyas1974 might know.

If not, then I'd suggest just closing this PR and making sure we have an issue filed under milestone 7.0.0 tracking adding this back when resources are available.  CC @agocke

**Comment by MattGal** [comment]
> Looks like [dotnet/core-eng#13921](https://github.com/dotnet/core-eng/issues/13921) tracks availability. @aik-jahoda, do we have a recent update? Looks like the last info was 1.5 months ago, it's possible things have changed. Perhaps @MattGal or @ilyas1974 might know.
> 
> If not, then I'd suggest just closing this PR and making sure we have an issue filed under milestone 7.0.0 tracking adding this back when resources are available. CC @agocke

Looks like we've got about 10 machines on osx.1200.arm64 and 57 on osx.1200.arm64.open ready for your usage now, so please give it a shot and let us know how it goes.  

Right now, usage is very light but if a bunch of teams migrate stuff there at once (especially if a lot of the work crashes and has to upload dumps) we may find this to be insufficient, but right now the queues are practically idle so party on, and let me know (or the FR teams channel) if you hit any issues.

**Comment by danmoseley** [comment]
@AaronRobinsonMSFT  any thoughts about this one?
```
 Interop/DllImportAttribute/DllImportPath/DllImportPathTest/DllImportPathTest.sh [Details] [History] [Artifacts]
Exception Message
Return code:      1
Raw output file:      /tmp/helix/working/B43009A2/w/B2050A13/uploads/Reports/Interop.DllImportAttribute/DllImportPath/DllImportPathTest/DllImportPathTest.output.txt
Raw output:
BEGIN EXECUTION
/tmp/helix/working/B43009A2/p/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false DllImportPathTest.dll ''
System.DllNotFoundException: Unable to load shared library 'MovedNativeLib' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(libMovedNativeLib, 0x0001): tried: 'libMovedNativeLib' (no such file), '/usr/local/lib/libMovedNativeLib' (no such file), '/usr/lib/libMovedNativeLib' (no such file), '/private/tmp/helix/working/B43009A2/w/B2050A13/e/Interop/DllImportAttribute/DllImportPath/DllImportPathTest/libMovedNativeLib' (no such file), '/usr/local/lib/libMovedNativeLib' (no such file), '/usr/lib/libMovedNativeLib' (no such file)
   at Test.GetZero_PathEnv()
   at Test.Main(String[] args)
Expected: 100
Actual: 101
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/tmp/helix/working/B43009A2/p
> /private/tmp/helix/working/B43009A2/w/B2050A13/e/Interop/DllImportAttribute/DllImportPath/DllImportPathTest/DllImportPathTest.sh
Expected: True
Actual:   False
```

**Comment by danmoseley** [comment]
Looks like that's the only failure on macOS 12 ARM. There's also a crash in Mono but on x64.

**Comment by AaronRobinsonMSFT** [comment]
@danmoseley I've no idea on this one. I don't know of any work in this product area recently so will assume it is either something flakey on the OS side or perhaps something in the test revamp that has been going on. This test does have special scripting logic so it is possible some disturbance was introduced. Any thoughts on the above @jkoritzinsky?

**Comment by agocke** [comment]
This is obsolete now

**Pull request closed without merging**